src/cards/DynamicRating.js

```diff
@@ -0,0 +1,63 @@
+import React, { useState, useEffect } from 'react'
+import { setLocal } from "../services";
+import {RatingCategory, Label, Rating, WorkIconStyle, WlanIconStyle, CupIconStyle} from './CafeCardStyles.js'
+import WorkIcon from '../images/WorkAtmosphereY.png'
+import WorkIconDisabled from '../images/WorkAtmosphereN.png'
+import WlanIcon from '../images/WLANY.png'
+import WlanIconDisabled from '../images/WLANN.png'
+import CupIcon from '../images/CupY.png'
+import CupIconDisabled from '../images/CupN.png'
+
+export default function DynamicRating() {
+
+    const [activeWorkIndex, setActiveWorkIndex] = useState(null)
+    const [activeWlanIndex, setActiveWlanIndex] = useState(null)
+    const [activeCoffeeIndex, setActiveCoffeeIndex] = useState(null)
+    
+    useEffect(() => {
+        setLocal('userRating', [activeWorkIndex + 1, activeWlanIndex + 1, activeCoffeeIndex + 1])
+    }, [activeWorkIndex, activeWlanIndex, activeCoffeeIndex])
+
+    const iconsArray = new Array(5).fill('')
+
+    const handleToggleWorkIcon = (index) => {
+        setActiveWorkIndex(index)
+    }
+
+    const handleToggleWlanIcon = (index) => {
+        setActiveWlanIndex(index)
+    }
+
+    const handleToggleCoffeeIcon = (index) => {
+        setActiveCoffeeIndex(index)
+    }
+
+    return (
+        <>
+            <RatingCategory>
```

Comment:
In order to summarise the mapping and writing the function only once, you may adjust the icons' sizes in Sketch. (one styling for all icons)